### DISCUSSION TITLE
"All projects should have unique names"

### DISCUSSION THREAD

**Discussion by matthew-dean** [vitest-dev/vitest#3939] [Q&A] [answered]
What does this line in the documentation mean? Names where? Defined where? All the projects in my monorepo certainly have unique names defined by their package.json file. Yet, setting up `defineWorskspace` like the documentation suggests `'packages/*'` results in `Error: Project name "packages" is not unique. All projects in a workspace should have unique names.`

If I change the workspace to `packages/*/*` the error says that the directory name for the project is not unique. Which, again, it is. So what the heck is the output referring to?

**Comment by sheremet-va** [comment]
Can you show your project folder structure and vite/vitest config files.

Name is `test.name` in the config file or a folder name, Vitest doesn't read `package.json`, it infers it from the folder.

**Comment by seawatts** [comment]
I'm seeing the same issue. I don't have a `vitest.config.ts` I only have a `vitest.workspace.ts`

```ts
// vitest.workspace.ts
export default ['packages/*'];
```

My project structure is 

```
root
  -> packages
      -> package-1
      -> package-2

vitest.workspace.ts
```

**Comment by matthew-dean** [comment, marked as answer]
I got this working! The Vitest workspace documentation is very opaque, and this error has nothing to do with the problem.

In my case:

1. Each package MUST have a vitest.config.ts, or at the VERY LEAST,
2. Each package that the workspace DEFINES must have a vitest.config.ts.

The way I get this working was to have a Vitest Workspace that looks like:
```ts
import { defineWorkspace } from 'vitest/config'
export default defineWorkspace([
  'packages/*/vitest.config.ts'
])
```
Essentially, Vitest will complain about names needing to be unique because it... I dunno, but whatever the Vitest documentation says is false. It won't use the package (folder) names and will fail.

**Reply by seawatts** [reply]
Great to know; thanks for looking into this.

You are correct, the docs state the following:

https://vitest.dev/guide/workspace.html#defining-a-workspace
```
Vitest will consider every folder in packages as a separate project even if it doesn't have a config file inside.
```

It seems like this is a bug and needs to be fixed. 
@sheremet-va can you confirm?

**Reply by sheremet-va** [reply]
I can’t confirm anything without a reproduction. Having projects without a config works on my side.

**Reply by Shakeskeyboarde** [reply]
@sheremet-va I believe the problem occurs when two workspace directories, have the same name, under different parent paths. For instance `packages/foo` has the same name as `tests/foo`. It doesn't matter if either has a vite/vitest config or not.

@matthew-dean You can fix the problem by actually adding a test configuration to one/both with an explicit name, so that it doesn't implicitly use the directory name as the test/project name.

```js
export default defineConfig({
  "test": {
    "name": "unique-name"
  }
});
```

**Comment by douglasg14b** [comment]
Also having this issue, except it seems to think that every package is duplicated when they are not, happening after upgrading to 2.1.0
